iphone/Classes/KrollContext.m

```diff
@@ -147,17 +147,8 @@ TiValueRef ThrowException(TiContextRef ctx, NSString *message, TiValueRef *excep
 
 static TiValueRef MakeTimer(TiContextRef context, TiObjectRef jsFunction, TiValueRef fnRef, TiObjectRef jsThis, TiValueRef durationRef, BOOL onetime)
 {
-  static dispatch_once_t timerInitializer;
-  static NSLock *timerIDLock = nil;
-  dispatch_once(&timerInitializer, ^{
-    timerIDLock = [[NSLock alloc] init];
-  });
-
   static double kjsNextTimer = 0;
```

Comment:
Maybe my lack of Obj-C expertise is shining through here, but shouldn't the static variable be declared/initialized somewhere else? Does this only set it to 0 the very first time? Or will every timer id == 0?

Reply:
It actually increases correctly, but you are right, it should rather be instance-scoped like [here](https://stackoverflow.com/a/1123095/5537752), although it doesn't really affect it here.